application/controllers/handler.js

```diff
@@ -1223,9 +1223,46 @@ module.exports = {
             console.log('Error', error);
             res(boom.badImplementation(error));
           });
+      }
+
+      case 2: { //video recorded
+        //check data values
+        if (!req.payload.data.fileName || !req.payload.data.deck || !req.payload.data.creationDate) {
+          return res(boom.badRequest('payload.data does not include all needed fields'));
+        }
 
-      default:
+        return userCtrl.find({_id: req.params.id})
+          .then((cursor) => cursor.toArray())
+          .then((array) => {
+            if (array.length !== 1) {
+              return res(boom.notFound('User does not exist'));
+            }
+
+            let email = array[0].email;
+            let videoURL = require('../configs/microservices').file.uri + '/video/' + req.payload.data.fileName;
+
+            let connectionPromise = util.sendEMail(email,
+              'New video about your live session of deck ' + req.payload.data.deck + (req.payload.data.revision) ? '-'+req.payload.data.revision : '',
+              'Dear ' + array[0].username + ',\n\nwe have finshed the video about your live session of deck ' + req.payload.data.deck + (req.payload.data.revision) ? '-'+req.payload.data.revision : '' + ' that ended at ' + req.payload.data.creationDate + '. Feel free to watch the video at: <a href="' + videoURL + '">' + videoURL + '</a>. if you want to download it, open the mentioned link, right click on the video and select "Save as ...". Please keep in mind we delete videos after eight weeks.\nFeel free to use, share and modify the video according to the license <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International (CC BY 4.0)</a>.\n\nYour SlideWiki Team');
+
+            return connectionPromise
```

Comment:
Lines 1248 to at least 1255 is duplicated code, as it appears in other cases too. Please move this outside of the switch case statement to avoid duplicated code.

Reply:
... Do you have an idea for me on how to do that? the switch statement does not return something, right?

Reply:
The time I review, it was duplicated code that may be moved to some into a function that is called in the switch-case statement. E.g. hand the text over to the function or specify codes and arguments that will choose the correct text automatically.